packages/fxa-auth-server/lib/payments/stripe.ts

```diff
@@ -911,13 +920,20 @@ export class StripeHelper {
    *
    * @param customer
    */
-  hasOpenInvoice(customer: Stripe.Customer) {
-    const subscription = customer.subscriptions?.data.find(
-      (sub) =>
-        ACTIVE_SUBSCRIPTION_STATUSES.includes(sub.status) &&
-        (sub.latest_invoice as Stripe.Invoice).status === 'open'
+  async hasOpenInvoice(customer: Stripe.Customer) {
+    const activeInvoices = customer.subscriptions?.data
+      .filter((sub) => ACTIVE_SUBSCRIPTION_STATUSES.includes(sub.status))
+      .map((sub) => sub.latest_invoice)
+      .filter((invoice) => invoice !== null);
+    if (!activeInvoices) {
+      return false;
+    }
+    const invoices = await Promise.all(
```

Comment:
Preferably we'd return true on the first invoice with an 'open' status instead of fetching all of them.